### ISSUE TITLE
Replace http client with opensearch-js client in dev tool console

### ISSUE THREAD

**Issue by zhongnansu** [opensearch-project/OpenSearch-Dashboards#3546]
`console` plugin doesn't interact with OpenSearch via opensearch-js client. It's just making http requests using Node.js http client. Given that the multiple datasource solution is mostly based on `opensearch-js` clients,  to onboard dev tool console with multiple datasource, we first need to refactor original logic to replace http client with opensearch-js client, and then extend it to use the data source client on demand.

Besides, keeping a unified way for OSD to interact with OpenSearch will benefit us long term. All visualizations/Discover retrieves data from OpenSearch through data plugin. And data plugin under the hood is using opensearch-js client.  I don't see the point for OSD dev tool console using http client to interact with OpenSearch. 

POC: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3544

Notes:
- To sending request from opensaearch-js client in dev tool console with user input, we can make generic request using `client.transport.request`
```
var client = new opensearch.Client();
client.transport.request({
  path: '/_search_with_clusters',
  method: 'PUT'
}).then(console.log.bind(console));
```

part of https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2699

**Comment by joshuarrrr** [comment]
@zhongnansu Was this completely resolved by #3544? If so, let's close.

**Comment by zhongnansu** [comment]
resolved but not merged to 2.7 branch yet

**Comment by joshuarrrr** [comment]
cool - we can close then - it's already in 2.x: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3639
